### ISSUE TITLE
 Split training questions to preserve level one thinking

### ISSUE THREAD

**Issue by ArneDeutsch** [daveshap/Raspberry#20]
Saw https://www.youtube.com/watch?v=zzaEBGOVKIg . If I understand correctly you plan to create training questions like this:

```
<INPUT>[User]QuestionThatRequiresThinking[/User]</INPUT>
<OUTPUT>[LM]WrongAnswer[/LM]
[LM]VerifyAnswer[/LM]
[LM]SecondTryWrong[/LM]
[LM]VerifySecondTry[/LM]
[LM]ThirdTryCorrect[/LM]
[LM]VerifyThirdTry[/LM]
</OUTPUT>
```
Doing it like this would probably train the LM to produce wrong answers at first, to then correct itself multiple times to ultimately come to the correct solution. This is probably not exactly what you want, because you harm the level one thinking system (the ability to create good answers on the first try).

I would suggest to split the thinking process into multiple training steps like this (note the placement of \<OUTPUT\> !!! ):

```
<INPUT>[User]QuestionThatRequiresThinking[/User]</INPUT>
<OUTPUT>[LM]CorrectAnswer[/LM]
[LM]VerifyAnswer[/LM]
</OUTPUT>
```

```
<INPUT>[User]QuestionThatRequiresThinking[/User]</INPUT>
[LM]WrongAnswer[/LM]
<OUTPUT>[LM]VerifyAnswer[/LM]
[LM]SecondTryCorrect[/LM]
[LM]VerifySecondTry[/LM]
</OUTPUT>
```

```
<INPUT>[User]QuestionThatRequiresThinking[/User]</INPUT>
[LM]WrongAnswer[/LM]
[LM]VerifyAnswer[/LM]
[LM]SecondTryWrong[/LM]
<OUTPUT>[LM]VerifySecondTry[/LM]
[LM]ThirdTryCorrect[/LM]
[LM]VerifyThirdTry[/LM]
</OUTPUT>
```

This way you would preserve the ability to produce high quality answers in the first try while implementing the verification loop in case an answer is not correct right away. Do not train the LM to produce bad answers in the first try, just train the verification/correction loop!

**Comment by daveshap** [comment]
This is not appropriate yet as we have no training data.

**Comment by JonIsHere242** [comment]
Hey this is just an idea but we could on the level-II thinking front what if we use a small cheap model like 4o-mini and then do tree of thought and use an integration model to rank the results on metrics like complexity correctness quality etc etc and get the response in integers between 1-100 in a format like '12-45-65' limit the tokens to some small number so that we save even more money and then use the synthetic metric data to make a tree of responses to maximize the exploration/exploitation of the tree. the we could do backprop on the responses to find the ones that are optimal and use that as the final synthetic data. Also recording the breadth/depth of the tree needed to make the highest quality response and tagging the prompts with tree params like a coding problem will be more narrow than an art problem.
          1
         /  \
        2    3
       / \     \
      4   5     6
         /       \
        7         8
                   \
                    9
